tiledb/sm/enums/datatype.h

```diff
@@ -347,6 +352,31 @@ inline bool datatype_is_time(Datatype type) {
       type == Datatype::TIME_AS);
 }
 
+inline uint8_t datatype_to_underlying(Datatype type) {
+  return static_cast<uint8_t>(type);
+}
+
+/* Note: Valid enumeration values are between 0 and 40. */
+inline Status datatype_is_valid(Datatype type) {
+  uint8_t datatype_enum = datatype_to_underlying(type);
+  if (datatype_enum > 40)
+    return Status_Error("Invalid Datatype " + std::to_string(datatype_enum));
+
+  return Status::Ok();
+}
+
+/* Note: Valid enumeration values are between 0 and 40. */
+inline Status datatype_str_is_valid(const std::string& datatype_str) {
+  Datatype datatype_type;
+  RETURN_NOT_OK(datatype_enum(datatype_str, &datatype_type));
+  uint8_t datatype_enum = datatype_to_underlying(datatype_type);
+
+  if (datatype_enum > 40)
```

Comment:
No need to repeat the test when you could also say `return is_datatype_valid(...);` here.